src/Analyser/MutatingScope.php

```diff
@@ -2329,25 +2330,43 @@ static function (Node $node, Scope $scope) use ($arrowScope, &$arrowFunctionImpu
 		}
 
 		if ($node instanceof FuncCall) {
-			if ($node->name instanceof Expr) {
+			$functionName = null;
+			if ($node->name instanceof Name) {
+				$functionName = $node->name;
+			} elseif ($node->name instanceof Expr) {
 				$calledOnType = $this->getType($node->name);
 				if ($calledOnType->isCallable()->no()) {
 					return new ErrorType();
 				}
 
-				return ParametersAcceptorSelector::selectFromArgs(
-					$this,
-					$node->getArgs(),
-					$calledOnType->getCallableParametersAcceptors($this),
-					null,
-				)->getReturnType();
+				if ($node->name instanceof String_) {
+					/** @var non-empty-string $name */
+					$name = $node->name->value;
+					$functionName = new Name($name);
+				} elseif (
+					$node->name instanceof FuncCall
+					&& $node->name->isFirstClassCallable()
+					&& $node->name->getAttribute('phpstan_cache_printer') !== null
+					&& preg_match('/\A(?<name>\\\\?[^()]+)\(...\)\z/', $node->name->getAttribute('phpstan_cache_printer'), $m) === 1
```

Comment:
Why the original code with:

```php
				return ParametersAcceptorSelector::selectFromArgs(
					$this,
					$node->getArgs(),
					$calledOnType->getCallableParametersAcceptors($this),
					null,
				)->getReturnType();
```

does not work here? That should be addressed instead.